packages/store/src/store.ts

```diff
@@ -84,7 +98,50 @@ export class Store {
    */
   selectSignal<T>(selector: TypedSelector<T>): Signal<T> {
     const selectorFn = this.getStoreBoundSelectorFn(selector);
-    return computed<T>(() => selectorFn(this._stateStream.state()));
+    const initialValue = selectorFn(this._stateStream.value);
+    const observable = this.selectFromStateStream(
+      // https://github.com/ngxs/store/issues/2180
+      // This is explicitly piped with the `debounceTime` to prevent synchronous
+      // signal updates. Signal updates occurring within effects can lead to errors
+      // stating that signal writes are not permitted in effects. This approach helps
+      // decouple signal updates from synchronous changes, ensuring compliance with
+      // constraints on updates inside effects.
+      // Developers should never rely on manually reading the signal after the state
+      // has been updated, whether synchronously or asynchronously, since the expected
+      // result may not be immediately available. To retrieve the current slice of the
+      // state, use `selectSnapshot` instead of directly accessing the signal. Signals
+      // are intended for use in templates or effects, as they always guarantee
+      // consistency with the latest signal value.
+      this._selectableStateStream.pipe(debounceTime(0, asapScheduler)),
+      selectorFn
+    );
+
+    try {
+      assertInInjectionContext(() => {});
+      return toSignal(observable, { initialValue });
+    } catch {
```

Comment:
Why is this part necessary? 
Isn't it enough just to be explicit with the injector?
Was there something that failed?

Reply:
Not enough, because the store captures the "very root" injector, which doesn't resolve `DestroyRef`s tied to a component.

Reply:
Basically, when you do `selectSignal` in a AppComponent, it's using the root injector, which doesn't resolve the AppComponent's DestroyRef.

Reply:
Well, since the `injectDestroyRef` internal function is calling `getLView`, which is referring to the current LView being saved on the stack, I think I might be wrong... but just wanted to double check that exact scenario.

Reply:
What happens if you call `runInIjectionContext` and you are already in an injection context? 
Does it give you the same one? Maybe this would be the solution?

Reply:
`runInInjectionContext` requires the injector to be provided, which will be the same root injector.